### PR TITLE
Deprecate ValueRecorder in favor of Histogram

### PR DESCRIPTION
**Pull request by hdost**
The Metric API Spec is now stable and ValueRecorder was replaced with
Histogram.

### CODE REVIEW

**Comment by hdost** [comment]
@TommyCpp I know the codecov are optional is it ok in this case since it's bascially just replacing a type?

**Comment by TommyCpp** [comment]
> @TommyCpp I know the codecov are optional is it ok in this case since it's bascially just replacing a type?

Yeah, don't worry about it especially for -0.02% changes

File: opentelemetry/src/metrics/histogram.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        self
+    }
+
+    /// Set the unit for this `Histogram`.
+    pub fn with_unit(mut self, unit: Unit) -> Self {
+        self.descriptor.config.unit = Some(unit);
+        self
+    }
+
+    /// Creates a new value recorder.
```
Comment (TommyCpp) [inline opentelemetry/src/metrics/histogram.rs:89]:
```suggestion
    /// Tries to create a new `Histogram`.
```

File: opentelemetry/src/metrics/histogram.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    pub fn try_init(self) -> Result<Histogram<T>> {
+        let instrument = self.meter.new_sync_instrument(self.descriptor)?;
+        Ok(Histogram(SyncInstrument::new(instrument)))
+    }
+
+    /// Creates a new value recorder.
```
Comment (TommyCpp) [inline opentelemetry/src/metrics/histogram.rs:95]:
```suggestion
    /// Creates a new `Histogram`.
```

**Comment by hdost** [comment]
Only change was a rebase

**Comment by TommyCpp** [comment]
Looks good. Thanks

**Pull request merged**